src/verifier.js

```diff
@@ -228,12 +222,10 @@ function verifyToken(
     }
 
     // Validate type
-    validateClaimType(values, claim, array, type === 'date' ? 'number' : 'string')
+    validateClaimType(values, claim, array, arrayValue, type === 'date' ? 'number' : 'string')
```

Comment:
there a bit more refactoring I would do here, because the array detection/conversion is a bit redundant and confusing. I appreciate that this logic is used in both the type validation and the value validation below, but let's at least give this `arrayValue` a better name, e.g. `isArray`, both here and in the argument names of the functions it's passed to.